Update to Lightning API 3.x

csirmazbendeguz opened the pull request:
In this branch I rewrote the failing test:
https://github.com/acquia/lightning-layout/pull/48

Commit:
Use 3.x lightning api branch

Commit:
Use 3.x lightning api branch

Commit:
Empty commit

Commit:
Revert "Use 3.x lightning api branch"

This reverts commit 68d68c9069b81f6fef789125d2802da3c59f00cd.

Commit:
Merge branch '8.x-3.x' into use-new-lightning-api

Commit:
Updated composer.lock

balsama commented:
Thanks! I think we need a pretty detailed entry in the CHANGELOG too. Something like this:

Lightning API now defaults to the 2.x branch of the JSON:API module. [See the release notes for the 2.x branch](https://www.drupal.org/project/jsonapi/releases/8.x-2.0-beta1). If you must use the 1.x branch, you can switch to it with the following commands:

```
composer require drupal/lightning_api:^2.7 --no-update
composer update drupal/lightning_api --with-all-dependencies
```

or by adding the following to the require section of your root composer.json file and running `composer update`:

```
drupal/lightning_api:^2.7
```

I also don't see a release of the 3.x of Lightning API. So let's get that done fist (8.x-3.0). For the record, I feel very comfortable calling it stable even though JSON:API 2.x is still RC.

Summary:
1. Create 3.0 release of Lightning API
2. Add the CHANGELOG to the PR

Commit:
Merge branch '8.x-3.x' into use-new-lightning-api

Commit:
Updated composer.lock

csirmazbendeguz commented:
I think the CHANGELOG should only be modified when creating a new release.

Currently it looks like this:
```
## 3.2.2
* Updated Lightning Core to 3.2, which security updates Drupal core
  to 8.6.2.

## 3.2.1
* Updated Lightning Media to 3.1.
...
```

If we added a new entry under 3.2.3 now, it would be confusing, because there is no 3.2.3 release yet.

csirmazbendeguz commented:
By the way, Lightning API currently defaults to the 2.x branch.
Should it default to 3.x?

Commit:
Update lightning_api to 3.x-dev

Commit:
Add api fixture context to behat

phenaproxima commented:
I'm OK with merging this. @balsama, do you concur?

balsama commented:
I think we should add the change log entry in this PR. I appreciate Bendis
point, but we’ve made adding the CR on commit standard.

And we should release 3.2.3 today anyway.
On Wed, Dec 19, 2018 at 8:31 AM Adam <notifications@github.com> wrote:

> I'm OK with merging this. @balsama <https://github.com/balsama>, do you
> concur?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/acquia/lightning/pull/620#issuecomment-448615759>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABdyjicF6qjBb3j5PDt7YVpKQEFqg98Fks5u6k3PgaJpZM4YN_cs>
> .
>
-- 
Adam Balsam Engineering Manager, Drupal | Acquia adam.balsam@acquia.com
53 State St., 10th Floor, Boston, MA 02109

Commit:
Update CHANGELOG

phenaproxima commented on CHANGELOG.md:
````diff
@@ -1,3 +1,21 @@
+  composer update drupal/lightning_api --with-all-dependencies
+  ```
+
+  or by adding the following to the require section of your root composer.json file and running 
````
Let's remove the last part of this (everything after the `composer require` and `composer update` commands); it's redundant. Otherwise, I wanna merge it!

Commit:
Remove redundant part from CHANGELOG, move full stop out of link

phenaproxima commented:
Looks great. Since the only new changes are to CHANGELOG and the last build is green, I'm merging this.
